File: MatchingTeam/app/src/main/java/com/example/matchingteam/RegisterUserApi.java
package com.example.matchingteam;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface RegisterUserApi {
    @POST("/api/users")
    Call<User> saveUser(@Body User user);
}

File: MatchingTeam/app/src/main/java/com/example/matchingteam/api/user/RegisterUserApi.kt
package com.example.matchingteam.api.user

import com.example.matchingteam.dto.user.RegisterUserDto
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface RegisterUserApi {
    @POST("/api/users")
    fun saveUser(@Body registerUserDto: RegisterUserDto): Call<Long>

    @GET("/api/authenticate")
    fun userEmailAuthentication(@Query("address") address: String): Call<String>

    @POST("/api/authenticate")
    fun userEmailAuthenticationConfirm(@Body authenticateCode: Int): Call<Boolean>
}